src/pages/workspaces/workspace/Data.js

```diff
@@ -549,33 +546,23 @@ class EntitiesContent extends Component {
   }
 }
 
-const DeleteObjectModal = class DeleteObjectModal extends Component {
-  constructor(props) {
-    super(props)
-    this.state = { deleting: false }
-  }
+const DeleteObjectModal = ({ name, workspace: { workspace: { namespace, bucketName } }, onSuccess, onDismiss }) => {
+  const [deleting, setDeleting] = useState(false)
 
-  delete = _.flow(
-    withErrorReporting('Error deleting object'),
-    Utils.withBusyState(v => this.setState({ deleting: v }))
-  )(async () => {
-    const { name, workspace: { workspace: { namespace, bucketName } }, onSuccess } = this.props
+  const doDelete = withErrorReporting('Error deleting object')(async () => {
+    setDeleting(true)
```

Comment:
Ah, the reason I remove `withBusyState` was because it was throwing an error in the console for trying to update state on an unmounted component. Good call, though.